codec-http2/src/test/java/io/netty/handler/codec/http2/Http2ServerUpgradeCodecTest.java

```diff
@@ -75,8 +75,11 @@ private static void testUpgrade(Http2ConnectionHandler handler, ChannelHandler m
         } else {
             codec = new Http2ServerUpgradeCodec((Http2FrameCodec) handler, multiplexer);
         }
-        assertTrue(codec.prepareUpgradeResponse(ctx, request, new DefaultHttpHeaders()));
-        codec.upgradeTo(ctx, request);
+        channel.eventLoop().execute(() -> {
+            assertTrue(codec.prepareUpgradeResponse(ctx, request, new DefaultHttpHeaders()));
+            codec.upgradeTo(ctx, request);
```

Comment:
we need to call this on the `EventLoop` as the codec asserts that we are on the event loop.